crawler.py

```diff
@@ -76,75 +77,104 @@ def close_all_windows_except_first(driver):
+                child_attributes = get_all_attributes(driver, child)
+
+                # If the element is not displayed or is disabled, the user can't interact with it. Skip
+                # non-displayed/disabled elements, since we're trying to mimic a real user.
+                if not child.is_displayed() or not child.is_enabled() or child in not_clickable_elems:
+                    continue
+
+                elem = child
                 break
 
-    close_all_windows_except_first(driver)
+            if elem is None:
+                return None
+
+            driver.execute_script('return arguments[0].scrollIntoView();', elem)
+            time.sleep(1)
+
+            if elem.tag_name in ['button', 'a']:
+                elem.click()
+            elif elem.tag_name == 'input':
+                input_type = elem.get_attribute('type')
+                if input_type == 'url':
+                    elem.send_keys('http://www.mozilla.org/')
+                elif input_type == 'text':
+                    elem.send_keys('marco')
+                elif input_type == 'email':
+                    elem.send_keys('prova@email.it')
+                elif input_type == 'password':
+                    elem.send_keys('aMildlyComplexPasswordIn2017')
+                elif input_type == 'checkbox':
+                    elem.click()
+                elif input_type == 'number':
+                    elem.send_keys('3')
+                elif input_type in ['submit', 'reset', 'button']:
+                    elem.click()
+                elif input_type == 'color':
+                    driver.execute_script("arguments[0].value = '#ff0000'", elem)
+                elif input_type == 'search':
+                    elem.clear()
+                    elem.send_keys('quick search')
+                elif input_type == 'radio':
+                    elem.click()
+                elif input_type == 'tel':
+                    elem.send_keys('1234567890')
+                elif input_type == 'date':
+                    elem.send_keys('20000101')
+                else:
+                    raise Exception('Unsupported input type: %s' % input_type)
+            elif elem.tag_name == 'select':
+                for option in elem.find_elements_by_tag_name('option'):
+                    if option.text != '':
+                        option.click()
+                        break
+            elif elem.tag_name in ['dialog', 'frame']:
+                if len(elem.find_elements_by_tag_name('button')) != 0:
+                    for button in elem.find_elements_by_tag_name('button'):
+                        button.click()
+                        break
+                elif len(elem.find_elements_by_tag_name('a')) != 0:
+                    for link in elem.find_elements_by_tag_name('a'):
+                        if link.text != '':
+                            link.click()
+                            break
+            not_clickable_elems.clear()
```

Comment:
We don't need to clear it, as we are going to return a few lines after and so the set will be cleared automatically